static/absolute/js/searchPage.js

```diff
@@ -41,54 +54,67 @@ window.onload = function () {
           location: [lat, long],
         }),
       });
-      data = await resp.json();
+      postResponse = await resp.json();
     } catch (err) {
       console.log(err);
       alert("error posting to api");
       return;
     }
 
-    if (data.status !== 200) {
+    if (postResponse.status !== 200) {
       // TODO(ved): How should we display errors?
-      console.log(data.error);
+      console.log(postResponse.error);
       alert("error posting to api");
       return;
     }
 
+    //Data was submitted successfully. We send a message to the server to then update each client with the same event ID.
+    socket.emit("data submitted", eventId);
     nameInput.value = "";
     addressInput.value = "";
   });
+
   document.getElementById("participants-btn").addEventListener("click", () => {
-    const id = getEventId();
-    window.location.href = `${window.location.origin}/${id}/participants`;
+    const eventId = getEventId();
+    window.location.href = `${window.location.origin}/${eventId}/participants`;
   });
 };
 
+function encodeAddress(address) {
+  const formattedAddress = encodeURIComponent(address)
+    .replace("!", "%21")
+    .replace("*", "%2A")
+    .replace("'", "%27")
+    .replace("(", "%28")
+    .replace(")", "%29");
+  return formattedAddress;
+}
+
 function getPosition(options) {
   return new Promise((resolve, reject) =>
     navigator.geolocation.getCurrentPosition(resolve, reject, options)
   );
 }
 
-/** Some Hard-coded data to display as restaurants temporarily. */
-const restaurant1 = {
-  name: "Nice Cafe",
-  address: "123 N. Street st.",
-  latLong: { lat: 37.0, lng: -122.0 },
-  rating: 5,
-  priceLevel: 2,
-  openingHours: "8:00am - 10:00pm",
-};
-const restaurant2 = {
-  name: "Bad Cafe",
-  address: "987 S. Street st.",
-  latLong: { lat: -37.0, lng: 122.0 },
-  rating: 2,
-  priceLevel: 4,
-  openingHours: "12:00am - 9:00am",
-};
+//This message is recieved from the server indicating that information has been submitted, and the client needs to refresh.
+socket.on("refresh", () => {
+  console.log(
+    "Refresh message received from server via Socket.io. Refreshing restaurant results."
+  );
+  restaurantSearch();
+});
 
-const allRestaurants = [restaurant1, restaurant2];
+async function restaurantSearch() {
```

Comment:
nit: in almost all programming languages we prefer function/method names to start with verbs. this makes it easy to distinguish references to functions vs references to variables. if I saw a reference to "restaurantSearch" without context, I would generally assume this is a variable holding args for some restaurant search. Prefer searchRestaurants() or something along those lines.

Reply:
Done. Changed to 'searchRestaurants'.